src/main/java/com/googleinterns/zoomtube/servlets/TranscriptLanguageServlet.java

```diff
@@ -0,0 +1,142 @@
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.ImmutableList;
+import com.google.gson.Gson;
+import com.googleinterns.zoomtube.data.TranscriptLanguage;
+import com.googleinterns.zoomtube.transcriptParser.TranscriptParser;
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.util.List;
+import java.util.Optional;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.apache.http.client.utils.URIBuilder;
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+import org.w3c.dom.NodeList;
+
+/** Provides information on a lecture. */
+public class TranscriptLanguageServlet extends HttpServlet {
+  /* Properties of the API Url. */
+  private static final String API_URL = "http://video.google.com/timedtext";
+  private static final String API_PARAM_TYPE = "type";
+  private static final String API_TYPE_LIST = "list";
+  private static final String API_PARAM_VIDEO = "v";
+  /* Error messages for missing or invalid video link. */
+  private static final String ERROR_MISSING_LINK = "Missing link parameter.";
+  private static final String ERROR_INVALID_LINK = "Invalid video link.";
+  /* Attributes for parsing the XML. */
+  private static final String TAG_TRACK = "track";
+  private static final String ATTR_LANG_CODE = "lang_code";
+  private static final String ATTR_NAME = "name";
+  private static final String ATTR_LANG_TRANSLATED = "lang_translated";
+
+  /* Name of input field used for lecture video link in lecture selection page. */
+  @VisibleForTesting static final String PARAM_LINK = "link-input";
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    Optional<String> error = validateGetRequest(request);
+    if (error.isPresent()) {
+      // TODO: Display a message in the client.
+      response.sendError(HttpServletResponse.SC_BAD_REQUEST, error.get());
+      return;
+    }
+
+    String videoUrl = request.getParameter(PARAM_LINK);
+    Optional<String> videoId = LectureServlet.getVideoId(videoUrl);
+    if (!videoId.isPresent()) {
+      // TODO: Display a message in the client.
+      response.sendError(HttpServletResponse.SC_BAD_REQUEST, ERROR_INVALID_LINK);
+      return;
+    }
+
+    URL transcriptLanguagesUrl = getTranscriptLanguagesUrlForVideo(videoId.get());
+    Document transcriptLanguagesDocument =
+        TranscriptParser.fetchUrlAsXmlDocument(transcriptLanguagesUrl);
+    List<TranscriptLanguage> transcriptLanguages =
+        parseTranscriptLanguages(transcriptLanguagesDocument);
+    writeTranscriptLanguages(response, transcriptLanguages);
+  }
+
+  /**
+   * Returns {@code Optional.empty()} if the video link is not missing. Else,
+   * returns an Optional containing an error message if it is.
```

Comment:
This comment is a bit weird maybe try eliminating "if it is"

Reply:
gotcha, added missing.